fix: select with header component can not scroll on android

qwang1113 opened the pull request:
1. fix select with header component can not scroll on android
2. use built-in overlay instead of react-native-portalize

qwang1113 committed:
fix: select with header component can not scroll on android & use built-in overlay instead of react-native-portalize

originalix committed:
chore: update hardware-js-sdk version (#1006)

originalix committed:
fix: check bonded android device & android search device failed OK-10405 OK-10408 (#1008)

* fix: check bonded android device

* fix: android search device interval

Co-authored-by: ll__ <loatheb.zhao@gmail.com>

revan-zhang committed:
feat: add deeplink for ios/android validation (#1012)

revan-zhang commented:
这个可以稍微晚点合入，我再拉下来测试一下，之前版本的 portalize 方案和 modalize 方案，总感觉整体性能和体验都有一些问题，可能这个需求需要再仔细调研一下

franco-chan committed:
style: WalletAvatar status indicator (#1010)

Commit:
feat: new history api (#900)

* feat: new history api

* fix: nagative transfer amount

Co-authored-by: kwoktung <onlyif2009@gmail.com>
Co-authored-by: ll__ <loatheb.zhao@gmail.com>

revan-zhang committed:
chore: add ipad detector & supportedNFC flag (#1015)

* chore: add ipad detector & supportedNFC flag

* fix: force web title

* chore: update version v3.1.0

originalix committed:
fix: check the devices that have associated wallets (#1013)

sunnylqm committed:
feat: replace chart component (#1005)

* feat: use new value-chart

* feat: cache chart data

* feat: optimise chart data

* fix: android label fresh issue

* fix: show current price when lose focus

* chore: cleanup

* chore: cleanup

* fix: web chart display

* fix: lint error

* feat: update app chart loading style

* feat: adjust chart loading style

linleiqin committed:
feat:add check nfc permission method (#1014)

taimanhui committed:
feat: merge AddImportedAccount & AddWatchAccount modals (#1011)

To handle complicated cases of various user inputs.

Also split check address and check watching account id functions, as not
every address of all networks can be added as a watching account,
typical cases are UTXO model networks.

Fixes: OK-9504 OK-9167

Commit:
fix: ext manifest (#1017)

* fix: ext manifest

* fix: service name cache

Commit:
feat: new history api (#1016)

* feat: new history api

* fix: lint

* fix: nonce number parse

* fix: lint

taimanhui committed:
fix: add existing wallet input validation problem on chrome (#1021)

ByteZhang1024 committed:
fix: android device search ui (#1020)

sunnylqm committed:
feat: add extreme label for chart (#1018)

Commit:
fix: ext onboarding from dapp OK-10216 (#1022)

* fix: hardware onboarding close window on ext

* fix: bugs

* fix: lint

revan-zhang committed:
feat: add isNativeIOSPadStore flag for platform env (#1024)

Commit:
Fix/history bugs (#1026)

* fix: history simple db storage

* fix: add history debug fields

revan-zhang committed:
fix: reload after account has created & at memo at UI components (#1027)

* chore: update react-native-collapsible-tab-view version & fix null for navigation issue

* fix: remove duplicate layout detection

* fix: reload after account has created & at memo at UI components

originalix committed:
fix: prevent duplicate loading to cause hardware error (#1028)

taimanhui committed:
fix: correct txid of near decodedTx (#1030)

Fixes: OK-10512

kwoktung committed:
feat: swap history (#1025)

ByteZhang1024 committed:
feat: Optimize hardware popup (#1019)

* feat: Optimize hardware popup

* fix: button popup

taimanhui committed:
fix: adapt BTC history to new mechanism  OK-10050 OK-10103 OK-10489 OK-10493 OK-10513 (#1029)

* fix: adapt BTC history to new mechanism

Fixes: OK-10050 OK-10103 OK-10489 OK-10493

* feat: hide speed up/cancel button for not-supported networks

Fixes: OK-10513

taimanhui committed:
fix: correct BTC transfer fee display (#1023)

Fixes: OK-10092

kwoktung committed:
refactor: useAppNavigation hook (#1031)

ByteZhang1024 committed:
fix: android database single row limit (#1032)

Commit:
fix: covalent tx parse (#1033)

taimanhui committed:
fix: use exact BTC sending amount for out transactions (#1035)

revan-zhang committed:
fix: reload accounts at initlal (#1036)

ByteZhang1024 committed:
chore: Optimize some error alerts (#1038)

sunnylqm committed:
feat: replace cgk apis and fix chart bugs OK-10502 OK-10549 (#1037)

* feat: replace cgk api

* fix: chart width

* chore: update flipper

* fix: chart price inconsistency

* feat: add lru cache for price api

* fix: price api cache

revan-zhang committed:
chore: add drop console statement at code compress (#1039)

sunnylqm committed:
fix: token detail header layout (#1040)

* fix: token detail header layout

* fix: remove header horizontal padding

* fix: token header layout

ByteZhang1024 committed:
fix: android error alerts (#1041)

franco-chan committed:
fix: i18n (#1042)

linleiqin committed:
feat:update discover list (#1045)

kwoktung committed:
chore: swap styles (#1047)

* fix: update locale

* chore: swap styles

* style: tweat history list

Co-authored-by: zuozhuo <zuoyizhuo@live.com>

sunnylqm committed:
fix: chart crash due to time disorder (#1046)

revan-zhang committed:
chore: update v3.2.0 version

taimanhui committed:
feat: show warning if sending to contract address (#1043)

Fixes: OK-8221

taimanhui committed:
fix: add missing "Learn more" translation (#1048)

Fixes: OK-10596

Commit:
chore: history ui (#1044)

* feat: history ui

* fix: lint

* fix: inpage webview isFocus check

* fix: add @providerApiMethod decorator for all provider api methods

taimanhui committed:
feat: add starcoin address history support (#1051)

Powered by the APIs from starcoin explorer.

Fixes: OK-10492

Commit:
chore: history detail (#1052)

* chore: update tx detail ui

* fix: update locale

kwoktung committed:
feat: add TypeWriter Component (#1054)

Co-authored-by: taimanhui <73929028+taimanhui@users.noreply.github.com>

taimanhui committed:
A couple of minor improvements (#1055)

* fix: use formValues.to in pre-send address check

* feat: include starcoin unknown transactions in history

sunnylqm committed:
fix: OK-9893 backup modal and update chart api  (#1056)

* fix: backup modal for testnet

* chore: update chart api

* chore: update deps

ByteZhang1024 committed:
chore: app upgrade (#1053)

linleiqin committed:
feat:NFT (#1034)

* feat:NFT

* feat:imrove get nft image source

* fix: lint error

taimanhui committed:
fix: address verification notice in presend form (#1058)

Fixes: OK-10680

ByteZhang1024 committed:
feat: hardware upload OK-10048 OK-10139 OK-10416 (#1050)

* feat: add basis hardware update feature

* feat: Optimize hardware update

* feat: Troubleshoot the screen lock problem

* chore: merge hardware upgrade

* feat: hardware status display

* chore: Optimized error notification

* fix: ios popup

* chore: remove check process

* chore: Roll back the downloaded firmware code

* chore: Normalize error messages

* feat: fix OneKey mini upgrade

* fix: ext error handle

* chore: change jssdk url

* chore: Optimize and upgrade UI

* chore: Fixes and cleanup code

ByteZhang1024 committed:
OK-10695 OK-10179 : fix desktop discover re-render problems (#1062)

* chore: Optimize dapp url input experience

* fix: Switching network tab route rerender

sunnylqm committed:
fix: avoid leading zeros in numberinput (#1064)

Commit:
fix: router bugs in firefox standalone window (#1063)

* fix: app speedUp and Cancel fee missing

* fix: route navigation bugs in firefox

* fix: lint

Commit:
Delete Expand.svg (#1061)

kwoktung committed:
OK-10724: fix swap error (#1065)

Commit:
fix: send address validation and some bugs OK-10135  OK-10368  OK-10503 OK-9529  (#1060)

* fix: bugs

* fix: lint

taimanhui committed:
fix: improve evm history event parsing (#1057)

- filter out only events related to the current account, limit its
  number to 10
- add a mutex lock for getting token info

Fixes: OK-10661

originalix committed:
OK-7107: confirm on hardware before receive (#1059)

* feat: hardware get address

* fix: splice btc path

* feat: confirm address on device

* feat: useEnsureConnected hook

* refactor: optimize useEnsureConnected hook

* fix: reset device confirm state

* chore: fetch locale

* feat: hardware wallet should confirm when receive

* chore: discard LocalValidationModal change

* feat: abort connect

* fix: style

* fix: repeat in hardware confirm

* feat: change toast type

* style: qrcode

* fix: lint

Co-authored-by: Franco <chenhuifan.gg@gmail.com>

Commit:
feat: send success receipt (#1066)

* fix: fix

* feat: add send feedback

linleiqin committed:
fix:nft list loading bug (#1067)

qwang1113 committed:
fix: OK-10610, OK-10665 (#1068)

linleiqin committed:
fix:remove haed code (#1069)

ByteZhang1024 committed:
fix: firmware upgrade error hint problem OK-10719 OK-10723 OK-10715 (#1070)

* fix: Desktop cross domain

* fix: ext create wallet error hint

* chore: Optimize upgrade i18n

* chore: Optimize Cancel Hardware popup

Commit:
feat: send fiat amount switch OK-10469  (#1071)

* fix: lint

* feat: fiat amount switch (desktop, ext only)

* feat: send amount fiat switch

* fix: lottie animations

taimanhui committed:
chore: update @onekeyfe/blockchain-libs to 0.0.37 (#1072)

ByteZhang1024 committed:
Feat: Touch v3/v4 sign & Android\iOS update firmware OK-10194 OK-10708 OK-10718 (#1073)

* fix: deviceUpdates is undefined

* chore: Update hardware SDK version

* chore: Remove native App update device skip check

franco-chan committed:
style: alert component (#1074)

Commit:
Fix/lottie view (#1077)

* fix: test

* fix: send amount button

* Revert "fix: test"

This reverts commit 900be6b80ff5e941f311711ccd62c88e59a564ef.

* fix: lottie view

* fix: static feedback icon

Commit:
fix: animation of feedback (#1080)

originalix committed:
fix: recover account cancel hardware process (#1076)

originalix committed:
OK-10609: abort ensure connect process (#1075)

* fix: abort connect process

* Merge branch 'onekey' into OK-10609

ByteZhang1024 committed:
fix: hardware operation popup (#1081)

Commit:
fix: next nonce of evm (#1086)

Commit:
Fix/fiat fetch (#1084)

* fix: fiat feth

* fix: ios lottie view bug

* fix: refresh fiat price interval

Commit:
fix: filling missing nonce of pending history (#1088)

linleiqin committed:
OK-10809 OK-10799 OK-10806:fix NFT issues (#1085)

ByteZhang1024 committed:
fix: Unable to start (#1087)

ByteZhang1024 committed:
Fix: ble upgrade OK-10716 OK-10723 (#1078)

* chore: Optimize the display of device details

* chore: Optimized error messages for confirm address

* feat: Display upgrade more error messages

* feat: update hardware sdk to v0.1.10

* chore: Optimize HardwarePopup

* fix: merge error

originalix committed:
fix: hardware wallets copy button text (#1091)

* feat: hardware wallets cannot copy address directly

* fix: set wallet type

kwoktung committed:
fix: help center styles (#1092)

linleiqin committed:
OK-10835 OK-10782 : fix NFT issues (#1094)

Commit:
fix: firefox dapp approve bugs (#1090)

ByteZhang1024 committed:
chore: Optimize native connect hradware connected hint (#1089)

sunnylqm committed:
feat: some optimisations (#1095)

linleiqin committed:
OK-10799:fix NFT video player on desktop (#1096)

revan-zhang committed:
chore: update v3.3.0 version

Commit:
fix: show near fee in history (#1099)

sunnylqm committed:
feat: update jsc and add getRandomValues shim (#1100)

* feat: use native impls

* fix: recover list back to blank page

* chore: remove stc bigint patch

* chore: revert native shims

* chore: revert ios change

* fix: ios deploy 14.0

Commit:
feat: copy tx info (#1097)

originalix committed:
fix: tips for not same wallet (#1102)

Commit:
Fix/firefox linking router OK-10845 (#1103)

* fix: firefox navigation linking

* fix: lint

qwang1113 committed:
fix: select in modal on android can not close select

qwang1113 committed:
fix: close select when click dapp webview on desktop
